kolena/utils.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021-2024 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import re
+from typing import Dict
+from typing import List
+
+import pandas as pd
+
+from kolena.annotation import LabeledTextSegment
+
+
+def _find_substring_indices(field_name: str, text: str, substrings: List[str], label: str) -> List[LabeledTextSegment]:
+    segments = []
+    for substring in substrings:
+        escaped_search = re.escape(substring)
+
+        pattern = rf"\b{escaped_search}\b"
+
+        matches = re.finditer(pattern, text, re.IGNORECASE)
+
+        segments.extend(
+            [
+                LabeledTextSegment(text_field=field_name, start=match.start(), end=match.end(), label=label)
+                for match in matches
+            ],
+        )
+
+    return segments
+
+
+def _extract_labeled_text_segments_from_keywords_single_row(
+    texts: Dict[str, str],
+    keyword_labels: Dict[str, List[str]],
+) -> List[LabeledTextSegment]:
+    labeled_segments = []
+    for label, keywords in keyword_labels.items():
+        for field_name, text in texts.items():
+            labeled_segments.extend(_find_substring_indices(field_name, text, keywords, label))
+    return labeled_segments
+
+
+def extract_labeled_text_segments_from_keywords(
+    df: pd.DataFrame,
+    text_fields: List[str],
+    keyword_labels: Dict[str, List[str]],
+    labeled_text_segments_column: str = "labeled_text_segments",
+) -> pd.DataFrame:
```

Comment:
What's the reason for including this function along with this PR?

Reply:
I am trying to provide a simpler way for the user to use the feature, instead of calculate the indexes and transform into our format, just provide us with keywords and tell us what label we should use. I think it could be useful when the user just want to highlight some keywords. 

This is part of the requirement as well
```
Setup a utility function to allow for string match use case
```

Reply:
Discussion continued in https://linear.app/kolena/issue/KOL-6082/ability-to-highlight-relevant-text-in-text-files-for-rag-evaluation#comment-b2e6f63f